client/src/pages/ScanStore.tsx

```diff
@@ -47,39 +91,259 @@ function ScanStore() {
     setShoppingList(sList);
   };
 
+  const setDebugItem = (barcode: string) => {
+    setCurBarcode(barcode);
+    setShoppingList(cartItems);
+  };
+
+  const addItem = async () => {
```

Comment:
All api fetching code show be outside react class ideally (can do it in a new issue)

Reply:
#56